drivers/storage/portworx/portworx.go

```diff
@@ -395,10 +396,45 @@ func (p *portworx) getCurrentMaxStorageNodesPerZone(
 	return 0, fmt.Errorf("storage disabled")
 }
 
+func isVsphere(cluster *corev1.StorageCluster) bool {
+	for _, env := range cluster.Spec.Env {
+		if env.Name == "VSPHERE_VCENTER" && len(env.Value) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func getCloudProvider(cluster *corev1.StorageCluster) string {
+	if isVsphere(cluster) {
+		return cloudops.Vsphere
+	}
+	// TODO: implement conditions for other providers
+	return ""
+}
+
 func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster) error {
 	if toUpdate.Annotations == nil {
 		toUpdate.Annotations = make(map[string]string)
 	}
+	if toUpdate.Spec.CloudStorage != nil {
```

Comment:
Can we move this under `setPortworxStorageSpecDefaults`? That way all the logic for setting cloud storage defaults stays in one place.